x-pack/solutions/observability/plugins/infra/server/lib/alerting/inventory_metric_threshold/inventory_metric_threshold_executor.ts

```diff
@@ -317,11 +318,16 @@ export const createInventoryMetricThresholdExecutor =
           ...additionalContext,
         };
 
+        const {
+          configuration: { metricAlias },
+        } = await libs.sources.getSourceConfiguration(savedObjectsClient, sourceId);
+
         const payload = {
           [ALERT_REASON]: reason,
           [ALERT_EVALUATION_VALUES]: evaluationValues,
           [ALERT_EVALUATION_THRESHOLD]: thresholds,
           [ALERT_GROUPING]: grouping,
+          [ALERT_CONTEXT]: { metricAlias },
```

Comment:
The ALERT_CONTEXT has been historically used to store additional context of alert (hosts, containers, etc.) before we added them at root level in alert doc. It was only used by one rule. But still to avoid any confusion, I'd suggest to use a new field such as `kibana.alert.index_pattern` to store index-pattern.

How are we storing index-pattern for other rules where we already added discover link (maybe it's not needed in other rules)?

Reply:
We also have a semi-related issue open, to list the indices/queries for rules in the Rule UI: https://github.com/elastic/kibana/issues/237592

Not sure where we're going with that, but I assume we'd arrange to have the rule types indicate indices and queries they are going to run, based on the current rule params.  Having that in place would make it straight-forward to add the index pattern to alerts for all rule types (I think).  cc: @jcger

Reply:
You mean like this, right?
https://github.com/elastic/kibana/pull/236880/commits/6c6a43096a70a55b81468f5aa266035406302d79

Actually this is better for another reason as well, for the log threshold alert I'm always getting the up to date index pattern, not the one that triggered the alert, so I should save the index pattern in the alert document for that rule as well.